projects/igniteui-angular/src/lib/data-operations/filtering-strategy.ts

```diff
@@ -109,3 +113,39 @@ export class FilteringStrategy extends BaseFilteringStrategy {
         return value;
     }
 }
+
+export class FormattedFilterStrategy extends FilteringStrategy {
+    constructor(public columns?: string[]) {
+        super();
+    }
+
+    public getFieldValue(rec: any, fieldName: string, isDate: boolean): any {
+        const value = resolveNestedPath(rec, fieldName);
+        let applyFormatter = false;
+        const column = this.grid.getColumnByName(fieldName);
+
+        if (!this.columns || this.columns.length === 0) {
+            applyFormatter = true;
+        } else if (this.columns.some(c => c === fieldName)) {
+            applyFormatter = true;
+        }
+
+        return applyFormatter && column.formatter ? column.formatter(value) : value;
+    }
+
+    public uniqueColumnFormattedValuesStrategy() {
+        return (column: IgxColumnComponent,
+            expression: IFilteringExpressionsTree,
+            done: (vals: any[]) => void) => {
+            this.getColumnData(column, expression, vals => done(vals));
+        };
+    }
```

Comment:
This method should not be part of the filtering strategy. It should be a separate function which performs the same operations as the `renderColumnValuesFromData` method of the ESF component.